code/API_definitions/kyc-age-verification.yaml

```
@@ -0,0 +1,598 @@
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details of how an API consumer requests an access token. Please refer to Identity and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the profile.
+
+    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the provider of the application consuming the API and the operator's API exposure platform, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
+
+    In cases where personal data is processed by the API and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of three-legged access tokens is mandatory. This ensures that the API remains in compliance with privacy regulations, upholding the principles of transparency and user-centric privacy-by-design.
+
+    # Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+
+
+  version: wip
+  contact:
+    email: sp-kyc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+servers:
+  - url: "{apiRoot}/kyc-ageverify/vwip"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+
+tags:
+  - name: Age Verify
+    description: Operations to verify the age of a user.
+
+paths:
+  /verify:
+    post:
+      tags:
+        - Age Verify
+      summary: Verify Age Threshold
+      description: |
+        Verify that the age of the subscriber associated with a phone number is equal to or greater than the specified age threshold value.
+
+        As it is possible that the person holding the contract and the person using the line may not be the same, the endpoint also admits a list of optional properties to be included in the request to improve the identification. The response may optionally include the `identityMatchScore` property with a value that indicates how certain it is that the information returned relates to the person that the API Client is requesting. To increase the reliability of the information returned, the operator may include in the response the `verifiedStatus` property, indicating whether the identity information in its possession has been verified against an official source.
+
+        The response may also include `contentLock` and `parentalControl` properties to indicate if the line has any kind of content filtering enabled.
+      operationId: verifyAge
+      security:
+        - openId:
+          - kyc-ageverify:ageverify
```

If we keep the `\verify` in paths then the scope should be:
```suggestion
          - kyc-age-verify:verify
```

---

Hi @rartych ,

Thanks.  I have asked to keep the original path 'ageverify', instead of 'verify'.  If url becomes either of kyc-ageverify or kyc-age-verify, 'ageverify' can be used, I understand.  Correct?

Thanks,
Toshi

---

There is no clear guideline but in section 3.1 there is such a sentence:

> When the POST method is used, the resource in the path must be a verb (e.g. retrieve-location and not location) to differentiate from an actual resource creation.

One can say that the verb is `verify`, so the kebab case should be used: `verify-age` or `age-verify` (but it starts with not a verb)

`"{apiRoot}/kyc-ageverify/vwip/ageverify"` looks redundant to me, but we can leave it for final review

---

Hi @rartych , @eric-murray , @fernandopradocabrillo ,

I had one question during my internal discussion regarding this part.

> Security:
>   -openId:
>       -kyc-ageverify:ageverify

(Currently the scope is kyc-ageverify:ageverify.)

Can we keep the first part 'kyc-ageverify', even though we change the filename to kyc-age-verification.yaml and change the url to {apiRoot}/kyc-age-verification/vwip ?  In another wey, should the first part 'kyc-ageverify' be aligned with 'api-name', i.e. kyc-age-verification?

Many thanks,
Toshi

---

As commented in the main conversation, "kyc-age-verification:verify" is now fine with us/KDDI.

Thanks,
Toshi